AbstractPage.java

```diff
@@ -0,0 +1,34 @@
+package pageobject;
+
+import lombok.Data;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.util.concurrent.TimeUnit;
+
+@Data
+public abstract class AbstractPage {
+
+    protected WebDriver webDriver;
+    protected WebDriverWait webDriverWait;
+    protected Actions actions;
+
+    public AbstractPage(WebDriver webDriver) {
+        this.webDriver = webDriver;
+        this.actions = new Actions(webDriver);
+        webDriverWait = new WebDriverWait(webDriver, 30);
+        webDriver.manage().window().maximize();
+        webDriver.manage().timeouts().implicitlyWait(30, TimeUnit.SECONDS);
+        PageFactory.initElements(webDriver, this);
+    }
+
+    public Actions moveOnWebElement(WebElement webElement) {
+        Actions builder = new Actions(webDriver);
```

Comment:
you have `actions ` object initialized in constructor, why do you need to create new instance here?

Reply:
I can do without creating a new instance here, that's right.